ethtest/test_case_splitter.go

```diff
@@ -113,7 +113,7 @@ func (s *TestCaseSplitter) SplitStateTests() (dividedTests []Transaction, err er
 				if fork == "Paris" {
 					fork = "Merge"
 				}
-				txCtx := newStateTestTxContext(stJson, msg, post, chainCfg, fork, postNumber)
+				txCtx := newStateTestTxContext(stJson, msg, post, chainCfg, fork, overall)
```

Comment:
The function `newStateTestTxContext` expects the post-number as its last argument and the resulting struct will store the given number as its `postNumber`.

Not sure what is the benefit of not using the `postNumber` here.

Assigning all tests a global number has conceptional issues:
- it makes the numbering dependent on the order in which files are enumerated by the list of json files
- adding, renaming, or removing a file will change the number of a test
- adding or removing post-cases of any test will change the number of subsequent tests

So the test number is not a very reliable bit of information, and it could not be used for e.g. filtering specific tests.

What is the motivation for this change?

Reply:
First thought when I implemented this was to number each `post` within `fork` within `file`
ie.
Test with path `/random/path/` contained two forks `Shangai` and `Cancun`. Each fork contained two `posts` (test cases).

So the intention was to find exactly the test that is failing using this number + path + fork.
Although this does not really work since there are tests that contain more tests within a file such as [this one](https://github.com/ethereum/tests/blob/develop/GeneralStateTests/Pyspecs/cancun/eip1153_tstore/reentrant_selfdestructing_call.json).

Adding the overall number helps understanding how many tests has been produced so far.

There is a `generatedTestHash` property for each test within the file which would distinguish each test within a file. We can add that but I feel like thats overcomplicating the logging? I believe knowing which file + which fork failed is enough. Wdyt?

Reply:
Sounds like the tuple `(path, test, post, fork)` where `test` is the index of the test in the file (which seems to be a better choice than the [name](https://github.com/ethereum/tests/blob/9201075490807f58811078e9bb5ec895b4ac01a5/GeneralStateTests/Pyspecs/cancun/eip1153_tstore/reentrant_selfdestructing_call.json#L2));

This order would correspond to the navigation path:
- first find the file
- within the file, find the test
- within the test, find the post block
- within the post block, find the fork

Reply:
The JSON converts into a map. We cannot guarantee the order of the tests.

Reply:
Sounds like sorting them by their key (which I guess is a string) would do the trick.